frontend/app/routes/prob-revshare.tsx

```diff
@@ -0,0 +1,272 @@
+          recipient until the page is closed or reloaded.
+        </HeadingCore>
+        <Card>
+          <Heading5>Wallet Address</Heading5>
+          <ShareInputHeader />
+          {shares.map((share, i) => {
+            return (
+              <ShareInput
+                key={i}
+                index={i}
+                name={share.name || ''}
+                onChangeName={(name) => handleChangeName(i, name)}
+                pointer={share.pointer}
+                onChangePointer={(pointer) => handleChangePointer(i, pointer)}
+                weight={share.weight || 0}
+                onChangeWeight={(weight) => handleChangeWeight(i, weight)}
+                weightDisabled={!share.pointer}
+                percent={
+                  totalWeight > 0 ? (share.weight || 0) / totalWeight : 0
+                }
+                percentDisabled={!share.pointer || shares.length <= 1}
+                onChangePercent={(percent) =>
+                  handleChangePercent(
+                    i,
+                    percent,
+                    share.weight || 1,
+                    totalWeight
+                  )
+                }
+                onRemove={() => handleRemove(i)}
+              />
+            )
+          })}
+          <hr />
+          <div className="flex flex-col-reverse md:flex-col gap-md">
+            {revSharePointers && (
+              <div className="flex h-[40px] items-center justify-between rounded-sm bg-interface-bg-main p-sm">
+                <CodeBlock
+                  link={revSharePointers}
+                  className="min-w-0 flex-1 overflow-x-auto whitespace-nowrap p-sm text-sm leading-normal"
+                />
+                <button
+                  onClick={handleCopyClick}
+                  aria-label={isCopied ? 'Copied' : 'Copy code to clipboard'}
+                >
+                  {isCopied ? (
+                    <SVGCheckIcon className="h-6 w-6" />
+                  ) : (
+                    <SVGCopyIcon className="h-6 w-6" />
+                  )}
+                </button>
+              </div>
+            )}
+            <div className="flex flex-col-reverse md:flex-row justify-end gap-xs">
+              <ToolsSecondaryButton
+                className="w-full md:w-auto"
+                onClick={() => setIsModalOpen(true)}
+              >
+                Import
+              </ToolsSecondaryButton>
+              <ToolsPrimaryButton
+                icon="share"
+                iconPosition="right"
+                className="flex w-full items-center justify-center md:w-auto"
+                onClick={addShare}
+              >
+                Add rev share
+              </ToolsPrimaryButton>
+            </div>
+          </div>
+        </Card>
+        <div className="my-lg md:my-md">
+          <RevShareChart shares={shares} />
+        </div>
+        <div className="flex flex-col gap-md mb-2xl">
```

Comment:
Consider extracting this into a separate component, as it doesn't need to re-render on any state change.